planner/core/exhaust_physical_plans.go

```diff
@@ -780,6 +780,18 @@ func (la *LogicalApply) exhaustPhysicalPlans(prop *property.PhysicalProperty) []
 	return []PhysicalPlan{apply}
 }
 
+func (p *LogicalWindow) exhaustPhysicalPlans(prop *property.PhysicalProperty) []PhysicalPlan {
+	if !matchItems(prop, p.ByItems) {
+		return nil
+	}
+	window := PhysicalWindow{
+		WindowFuncDesc: p.WindowFuncDesc,
+		PartitionBy:    p.PartitionBy,
+	}.Init(p.ctx, p.stats.ScaleByExpectCnt(prop.ExpectedCnt), &property.PhysicalProperty{ExpectedCnt: math.MaxFloat64})
```

Comment:
since the required child order property is empty, data will be sort inside the window operator according to the partition and order by items?

Reply:
The order operator is added in https://github.com/pingcap/tidb/pull/8630/files#diff-b9e75a99170ea35a67074d5d3d3305adR2535.

Reply:
looks like this physical window operator requires the child output to be sorted by `ByItems`, maybe it's better to set the required child property to `ByItems`? Hope this can prevent some potential bugs, for example, we forget to build a sort operator for this window operator in the plan building phase.

Reply:
But you cannot construct a property for `order by a desc, b asc`.

Reply:
Sadness, can we modify the physical property definition?

Reply:
Of course.

Reply:
Seems we cannot push down the property for `order by (a+1)`, need the physical property definition to be modified once more?